src/main/java/bookstore/service/UserServiceImpl.java

```diff
@@ -4,25 +4,38 @@
 import bookstore.dto.user.UserResponseDto;
 import bookstore.exception.RegistrationException;
 import bookstore.mapper.UserMapper;
+import bookstore.model.Role;
 import bookstore.model.User;
+import bookstore.repository.RoleRepository;
 import bookstore.repository.UserRepository;
 import jakarta.transaction.Transactional;
+import java.util.Optional;
+import javax.management.relation.RoleNotFoundException;
 import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 @Service
 @RequiredArgsConstructor
 @Transactional
 public class UserServiceImpl implements UserService {
     private final UserRepository userRepository;
+    private final RoleRepository roleRepository;
     private final UserMapper userMapper;
+    private final PasswordEncoder passwordEncoder;
 
     @Override
-    public UserResponseDto register(UserRegistrationRequestDto requestDto) {
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RoleNotFoundException {
         if (userRepository.existsByEmail(requestDto.getEmail())) {
             throw new RegistrationException("Email already exists: " + requestDto.getEmail());
         }
+        Optional<Role> defaultRole = roleRepository.findByRole(Role.RoleName.USER);
         User user = userMapper.toModel(requestDto);
+        user.setPassword(passwordEncoder.encode(requestDto.getPassword()));
+        user.getRoles().add(defaultRole
+                .orElseThrow(() -> new RoleNotFoundException("Can't find role: "
+                        + Role.RoleName.USER)));
```

Comment:
```suggestion
        User user = userMapper.toModel(requestDto);
        user.setPassword(passwordEncoder.encode(requestDto.getPassword()));
        Role defaultRole = roleRepository.findByRole(Role.RoleName.USER).orElseThrow(() -> new EntityFoundException("Can't find role: " + Role.RoleName.USER));;
        user.setRoles(Set.of(defaultRole));
```
you can use EntityNotFoundException